network-libp2p/src/network.rs

```diff
@@ -370,60 +361,54 @@ impl Network {
-            let transport = WebsocketTransport::default();
+            #[cfg(all(target_family = "wasm", not(feature = "tokio-websocket")))]
+            let transport = websocket_websys::Transport::default();
 
-            #[cfg(all(not(feature = "tokio-websocket"), not(feature = "wasm-websocket")))]
+            #[cfg(all(not(feature = "tokio-websocket"), not(target_family = "wasm")))]
             let transport = MemoryTransport::default();
 
-            let noise_keys = noise::Keypair::<noise::X25519Spec>::new()
-                .into_authentic(keypair)
-                .unwrap();
-
-            let mut yamux = yamux::YamuxConfig::default();
+            let mut yamux = yamux::Config::default();
             yamux.set_window_update_mode(yamux::WindowUpdateMode::on_read());
 
             Ok(transport
                 .upgrade(core::upgrade::Version::V1)
-                .authenticate(noise::NoiseConfig::xx(noise_keys).into_authenticated())
+                .authenticate(noise::Config::new(keypair).unwrap())
                 .multiplex(yamux)
                 .timeout(std::time::Duration::from_secs(20))
                 .boxed())
         }
     }
 
-    fn new_swarm(
+    async fn new_swarm(
         clock: Arc<OffsetTime>,
         config: Config,
         contacts: Arc<RwLock<PeerContactBook>>,
-        peer_score_params: PeerScoreParams,
-        executor: impl TaskExecutor + Clone + Send + 'static,
-    ) -> Swarm<NimiqBehaviour> {
-        let local_peer_id = PeerId::from(config.keypair.public());
-
+        peer_score_params: gossipsub::PeerScoreParams,
+    ) -> Swarm<behaviour::Behaviour> {
+        let keypair = config.keypair.clone();
         let transport =
-            Self::new_transport(&config.keypair, config.memory_transport, &config.tls).unwrap();
-
-        let behaviour = NimiqBehaviour::new(config, clock, contacts, peer_score_params);
+            Self::new_transport(&keypair, config.memory_transport, &config.tls).unwrap();
 
-        let limits = ConnectionLimits::default()
-            .with_max_pending_incoming(Some(16))
-            .with_max_pending_outgoing(Some(16))
-            .with_max_established_incoming(Some(4800))
-            .with_max_established_outgoing(Some(4800))
-            .with_max_established_per_peer(Some(MAX_CONNECTIONS_PER_PEER));
+        let behaviour = behaviour::Behaviour::new(config, clock, contacts, peer_score_params);
 
         // TODO add proper config
-        SwarmBuilder::with_executor(
-            transport,
-            behaviour,
-            local_peer_id,
-            Box::new(move |fut| {
-                executor.exec(fut);
-            }),
-        )
-        .connection_limits(limits)
-        .build()
+        #[cfg(not(target_family = "wasm"))]
+        let swarm = SwarmBuilder::with_existing_identity(keypair)
+            .with_tokio()
+            .with_other_transport(|_| transport)
```

Comment:
A big benefit of the new `SwarmBuilder` is that it enforces at compile-time, that you compose transports together correctly. For example, DNS needs to wrap all other transports (except relay).

By using `with_other_transport`, you are not leveraging any of this. I would suggest to refactor this into two functions, one of them for `wasm` and the other one `non-wasm` and use the entire `SwarmBuilder` in there, without any further conditionally-compiled code.